net-test/tests/net/url-string.rkt

```diff
@@ -137,7 +137,11 @@
   ;; and in relative paths as long as it's not in the first element
   (test-s->u #(#f #f #f #f #f (#("x") #("y:@z")) () #f)
              "x/y:@z")
-
+  
+  ;; test ipv6 support
+  (test-s->u #("https" #f "[1234:93bf::8]" #f #t (#("")) () #f)
+        "https://[1234:93bf::8]/")
+  
```

Comment:
Not only this lacks negative tests, but also doesn't include longer URLs with components after the IPv6 address.
It's important to have tests for those too, especially since IPv6 addresses are matched with the greedy `.*`. (I'd try adding a `]` after the end of the IPv6 address and see what happens compared to what should happen).